GIT_CONVENTIONS.md

```diff
@@ -0,0 +1,87 @@
+# Git Conventions for the ORAS Community
+
+This guide is a specification for adding human and machine readable meaning to Pull Request and git commit messages in the ORAS community. It provides an easy set of rules for creating an explicit commit history; which makes it easier to explore a more structured commit history and automatically generating CHANGELOGs for each new release. 
+
+When contributing to the ORAS projects, please first discuss the change you wish to make via a GitHub issue, a GitHub discussion, or any other methods with the ORAS community maintainers before making a change.
+
+> Note: Please follow the [Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md) in all your interactions with the ORAS project.
+
+
+## Commit Message Guidelines
+
+This guide provides very precise rules over how Pull Requests and git commit messages.
+
+### Commit Message Format
+Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+format that includes a **type**, a **scope** and a **subject**:
+
+```
+<type>[optional scope]: <subject>
+
+[optional body]
+
+[optional footer(s)]
+```
+The **header** is mandatory and the **scope** of the header is optional.
+
+The first line of commit messages are recommended to be 52 characters or less. The maximum length of body lines should not exceed 72 characters. This allows the message to be easier
+to read on GitHub as well as in various git tools.
+
+The footer should contain a [closing reference to an issue](https://help.github.com/articles/closing-issues-via-commit-messages/) if any.
+
+Samples: (even more [samples](https://www.conventionalcommits.org/en/v1.0.0/#examples))
+
+```
+docs: correct spelling of CHANGELOG
+```
+```
+fix: prevent racing of requests
+
+Introduce a request id and a reference to latest request. Dismiss
+incoming responses other than from latest request.
+
+Reviewed-by: Z
+Refs: #123
+```
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
```

Comment:
It might be helpful here to say something like:

```suggestion
If you need to undo a commit, you might want to use [revert](https://www.theserverside.com/tutorial/How-to-git-revert-a-commit-A-simple-undo-changes-example). We ask for a specific format for these reverted commits too. When the commit reverts a previous commit, it should begin with `revert: ` followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
```

Reply:
Reading further, if I missed `<type>` in the example, I'm not sure if this type section is relevant to just revert, or more generally for any commit? I would add a statement to give context for the reader, e.g., "When you write a commit message, be sure to include a tag that notes the *type* of change. I do see `<type>` in the example above (after the fact) but honestly I missed it the first time around!

Reply:
I will add a sentence in `Commit Message Format` section after the sample format.

```
The **header** is mandatory, and the **scope** of the header is optional. Commits MUST be prefixed with a type.
```

What do you think?

Reply:
Also I moved the Revert section to the end of Commit Message Guidelines section, this might help as well

Reply:
I think that will definitely help - the reader starts with the context of "this is about the structure of a commit message" so it will read nicely to have the revert topic mentioned after that. :)